apps/edit/app.lisp

```diff
@@ -76,32 +71,42 @@
 (defun-bind mouse-cursor (mouse_xy)
 	(defq buffer (elem text_buffer current_text))
 	(bind '(ox oy cx cy sx) (elem text_position current_text))
-
 	(defq cursor_xy (list cx cy) char_wh (vdu-char-size vdu) offset_xy (list ox oy))
 	(setq cursor_xy (map + (map / mouse_xy char_wh) offset_xy)
 		cx (elem 0 cursor_xy) cy (elem 1 cursor_xy))
-	(if (>= cy (length buffer)) (setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
+	(if (>= cy (length buffer)) 
+		(setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
 	(if (> cx (length (elem cy buffer)))
 		(setq cx (max (set-sticky) (length (elem cy buffer)))) (setq sx cx))
-
 	(elem-set text_buffer current_text buffer)
 	(elem-set text_position current_text (list ox oy cx cy sx)))
 
 (defun-bind open-buffer (path)
-	(defq index (length text_store) title path buffer (list (join " " (ascii-char 10))) pos (list 0 0 0 0 0))
-	(unless (or (eql path "") (not (catch (file-stream path) t)))
-		(setq buffer (list))
-		(each-line (lambda (_) (push buffer _)) (file-stream path)))
-	(if (eql title "") 
-		(setq title (cat "Untitled-" (str (setq tmp_num (inc tmp_num))))))
-	(push text_store (list index path title buffer pos))
+	(defq i 0 index (length text_store) pos (list 0 0 0 0 0))
+	(if (and (not (some (lambda (_) (eql "/" _)) path)) (not (eql path ""))) 
+		(setq path (cat home_dir path)))
```

Comment:
(some (lambda (_) (eql "/" _)) path)

You can do this check faster with (find "/" path), will return nil or the index of the char. It's like strrchar in C.

Chris